### ISSUE TITLE
Upgraded SDK to 3.2.162 and now Dev Cluster won't start

### ISSUE THREAD

**Issue by NArnott** [Azure/service-fabric-issues#1179]
I get events in the System event log:

The Service Fabric Host Service service terminated unexpectedly.  It has done this 30 time(s).  The following corrective action will be taken in 5000 milliseconds: Restart the service.

I tried uninstalling the SDK compeltely, and deleted the SfDevCluster folder, restarted machine and installed again. Same result.

**Comment by NArnott** [comment]
I noticed in the FabricDeployer trace the following regarding DISM. Is this related?

2018/07/17-16:50:45.083,Info,10940,FabricDeployer.FabricDeployer,Dism initializing...
2018/07/17-16:50:45.086,Info,10940,FabricDeployer.FabricDeployer,Dism initializing completed, opening online session...
2018/07/17-16:50:45.253,Info,10940,FabricDeployer.FabricDeployer,Dism opening online session completed, getting feature info...
2018/07/17-16:50:45.496,Info,10940,FabricDeployer.FabricDeployer,Dism feature Containers is in state: DismStateStaged
2018/07/17-16:50:45.496,Info,10940,FabricDeployer.FabricDeployer,Dism closing session...
2018/07/17-16:50:45.521,Info,10940,FabricDeployer.FabricDeployer,Dism closing session completed
2018/07/17-16:50:45.521,Info,10940,FabricDeployer.FabricDeployer,Dism shutting down...
2018/07/17-16:50:45.534,Info,10940,FabricDeployer.FabricDeployer,Dism shutting down completed
2018/07/17-16:50:45.538,Info,10940,FabricDeployer.FabricDeployer,Dism log file content: 
	2018-07-17 10:50:45, Warning               DISM   DISM Provider Store: PID=14672 TID=16816 Failed to load the provider: C:\WINDOWS\SYSTEM32\Dism\SiloedPackageProvider.dll. - CDISMProviderStore::Internal_GetProvider(hr:0x8007007e)
	2018-07-17 10:50:45, Warning               DISM   DISM Provider Store: PID=14672 TID=16816 Failed to load the provider: C:\WINDOWS\SYSTEM32\Dism\MetaDeployProvider.dll. - CDISMProviderStore::Internal_GetProvider(hr:0x8007007e)
	[14672] [0x8007007b] FIOReadFileIntoBuffer:(1381): The filename, directory name, or volume label syntax is incorrect.
	[14672] [0xc142011c] UnmarshallImageHandleFromDirectory:(641)
	[14672] [0xc142011c] WIMGetMountedImageHandle:(2897)
	[14672] [0x8007007b] FIOReadFileIntoBuffer:(1381): The filename, directory name, or volume label syntax is incorrect.
	[14672] [0xc142011c] UnmarshallImageHandleFromDirectory:(641)
	[14672] [0xc142011c] WIMGetMountedImageHandle:(2897)
	2018-07-17 10:50:45, Warning               DISM   DISM OS Provider: PID=12224 TID=13872 Unable to set the DLL search path to the servicing stack folder. C:\Windows may not point to a valid Windows folder. - CDISMOSServiceManager::Final_OnConnect
	2018-07-17 10:50:45, Warning               DISM   DISM Provider Store: PID=12224 TID=13872 Failed to load the provider: C:\WINDOWS\TEMP\850D9817-C839-41DE-B599-897D4DE89EE6\PEProvider.dll. - CDISMProviderStore::Internal_GetProvider(hr:0x8007007e)

**Comment by NArnott** [comment]
I was able to workaround the issue by using the FabricContainerAppsEnabled setting mentioned here: https://github.com/Azure/service-fabric-issues/issues/1056

I am not using containers, and don't have docker or HyperV installed on my machine. Is that the issue?

**Comment by CuddleBunny** [comment]
I've been having issues with my local cluster since the update as well. I haven't had much time to troubleshoot it yet. Changing `FabricContainerAppsEnabled` to false does not seem to work for me. Cluster was working just fine on the previous version.

**Comment by sdkadam2017** [comment]
Is there a way you can share fabric logs for the upgraded version?

**Comment by NArnott** [comment]
I'd love to. Which logs and how do I get them to you?

**Comment by dragav** [comment]
(assuming Windows) The reg key HKLM\SOFTWARE\Microsoft\ServiceFabric\FabricLogRoot should point to the location of the logs. Zip the directory and attach the file here; you can set a password on the zip and email it to me (dragosav at microsoft….). Alternatively, you could upload the archive to Azure storage, and email me a time-bound SAS key - I should be able to get through this in the next couple of days.

**Comment by NArnott** [comment]
@dragav Here's the log. I'll email you the password. 

[Log.zip](https://github.com/Azure/service-fabric-issues/files/2235893/Log.zip)

**Comment by dragav** [comment]
@NArnott Thanks, I replied via email. The logs point, indeed, to the same problem as reported by issue [1056](https://github.com/Azure/service-fabric-issues/issues/1056), and denotes missing permissions to the machine key store. Please try the suggested workaround and let me know - I'll keep this issue open until then.

**Comment by masnider** [comment]
@NArnott did this workaround work for you?

**Comment by NArnott** [comment]
@masnider I did the workaround mentioned in https://github.com/Azure/service-fabric-issues/issues/1056, and set FabricContainerAppsEnabled to false. Since we aren't using containers at this time, it seamed the easier workaround, and it's already proved to work.

Since both options are workarounds, I assume there is a fix in the works so it will work by default?

**Comment by masnider** [comment]
Yes we're looking to see if we can ensure that the correct permissions are present, but also to figure out what is causing it to be different across machines in the first place.
